### ISSUE TITLE
Implement usage of PI alias file

### ISSUE THREAD

**Issue by QuanMPhm** [CCI-MOC/process_csv_report#43]
Because the current invoice processing script may create undesired output if a PI uses multiple aliases, we want to give the invoice script an alias file for it to correctly bill PIs with multiple aliases. This requires two separate tasks:

- [ ] Creation and storage of an alias file in long-term storage(?). Our list of currently known aliases are below. Hopefully no more aliases will appear the the future
- [ ] Modify `process_report.py` to use the alias file

**Comment by QuanMPhm** [comment]
@knikolla I'll add an argument to `process_report.py` to accept a user-provided alias.csv file, with the option to also fetch the file from a hardcoded location in S3, similar to how the PI file will be fetched. 

Given an alias file, each time the invoice script applies the New-PI credit and makes the corresponding changes to the PI file, it will make sure changes to a PI's entry will be applied to the PI's aliases. 

To explain in more detail, as the invoice script [iterates through each billable PI](https://github.com/CCI-MOC/process_csv_report/blob/cd231b45086c8e5b97eae1e1f099b86759fc86b7/process_report/process_report.py#L349), determines how the New-PI credit should be applied to them, and modifies the PI's entry in the PI file, it make replicate that same modified entry across the PI's aliases.

This makes sure that in the (presumably) rare case where a PI appears in an invoice under more than 1 alias, the credits they use are still correctly tracked. 

@knikolla  Implementation of this alias file may be significantly more complex after #44 is merged. In which case #44 itself will need to be modified, or a new PR will be needed to update the alias file implementation?

**Comment by knikolla** [comment]
@QuanMPhm The new argument makes sense.

With regards to the updating logic: 

A much simpler solution is to have a step before you start processing the new PI credit where you replace all aliases with the canonical name. 

So if your script encounters `username1` as a PI, and there is a line in alias for `username1 -> email1`, you replace username1 with email1 in all encountered rows of the file. 

This makes it so you don't have to replicate any changes to the PI file and you have one canonical row for each PI in the PI csv file.

**Comment by QuanMPhm** [comment]
@knikolla Would we eventually remove alias entries from the PI file as they become out-of-date?

**Comment by knikolla** [comment]
@QuanMPhm alias entries will not become out of date as users can't change their usernames.
